frontend/packages/dev-console/src/components/edit-deployment/utils/edit-deployment-utils.ts

```diff
@@ -8,3 +28,415 @@ export const getContainerNames = (containers: { [key: string]: any }[]) => {
+  };
+};
+
+export const getStrategyData = (
+  type: string,
+  strategy: any,
+  resName: string,
+  resNamespace: string,
+) => {
+  switch (type) {
+    case DeploymentStrategyType.recreateParams: {
+      const { mid: midHook, post: postHook, pre: preHook, timeoutSeconds } =
+        strategy.recreateParams ?? {};
+      return {
+        data: {
+          recreateParams: {
+            timeoutSeconds,
+            pre: getLifecycleHookData(preHook),
+            mid: getLifecycleHookData(midHook),
+            post: getLifecycleHookData(postHook),
+          },
+        },
+        imageStreamData: {
+          pre: getLchImageStreamData(resName, resNamespace, preHook?.tagImages),
+          mid: getLchImageStreamData(resName, resNamespace, midHook?.tagImages),
+          post: getLchImageStreamData(resName, resNamespace, postHook?.tagImages),
+        },
+        lifecycleHooks: {
+          pre: !!preHook,
+          mid: !!midHook,
+          post: !!postHook,
+        },
+      };
+    }
+    case DeploymentStrategyType.customParams: {
+      const { command, environment, image } = strategy.customParams ?? {};
+      return {
+        data: {
+          customParams: {
+            command,
+            environment,
+            image,
+          },
+        },
+      };
+    }
+    case DeploymentStrategyType.rollingParams: {
+      const {
+        post: postHook,
+        pre: preHook,
+        timeoutSeconds,
+        updatePeriodSeconds,
+        intervalSeconds,
+        maxSurge,
+        maxUnavailable,
+      } = strategy.rollingParams ?? {};
+      return {
+        data: {
+          rollingParams: {
+            timeoutSeconds,
+            pre: getLifecycleHookData(preHook),
+            post: getLifecycleHookData(postHook),
+            updatePeriodSeconds,
+            intervalSeconds,
+            maxSurge,
+            maxUnavailable,
+          },
+        },
+        imageStreamData: {
+          pre: getLchImageStreamData(resName, resNamespace, preHook?.tagImages),
+          post: getLchImageStreamData(resName, resNamespace, postHook?.tagImages),
+        },
+        lifecycleHooks: {
+          pre: !!preHook,
+          post: !!postHook,
+        },
+      };
+    }
+    case DeploymentStrategyType.rollingUpdate: {
+      const { maxSurge, maxUnavailable } = strategy.rollingUpdate ?? {};
+      return { data: { rollingUpdate: { maxSurge, maxUnavailable } } };
+    }
+    default:
+      return {};
+  }
+};
+
+export const getStrategy = (
+  deployment: K8sResourceKind,
+  resourceType: string,
+): DeploymentStrategy => {
+  const { strategy } = deployment.spec ?? {};
+  let type: string;
+
+  if (resourceType === Resources.OpenShift) {
+    type = strategy?.type ?? DeploymentStrategyType.rollingParams;
+    return {
+      type,
+      lifecycleAction: LifecycleAction.execNewPod,
+      ...getStrategyData(type, strategy, deployment.metadata?.name, deployment.metadata?.namespace),
```

Comment:
is there any case where metadata will be undefined?
```suggestion
      ...getStrategyData(type, strategy, deployment.metadata.name, deployment.metadata.namespace),
```